third_party/amd/lib/TritonAMDGPUToLLVM/Utility.cpp

```diff
@@ -156,12 +156,23 @@ Value llGetPid(Location loc, RewriterBase &rewriter, ModuleOp moduleOp,
 }
 
 Value llLoad(RewriterBase &rewriter, Location loc, Value ptr, Type elemTy,
-             Value pred, Value falseVal, bool nt) {
+             Value pred, Value falseVal, triton::CacheModifier cm) {
   Type funcType = getFunctionType(elemTy, ValueRange({ptr, pred, falseVal}));
   auto parent = ptr.getParentRegion()->getParentOfType<LLVM::LLVMFuncOp>();
-  auto funcNameRaw = nt ? mlir::LLVM::AMD::Predicated_Load_NT
-                        : mlir::LLVM::AMD::Predicated_Load;
-  auto funcName = mangleFunc(funcNameRaw, funcType);
+  auto getLoadNameRaw = [](triton::CacheModifier cm) {
```

Comment:
What's the purpose of this extra functor? Just inlined switch statement works?

Reply:
I don't like the approach of creating uninitialized variables and init it in switch. In my sense such default initialization (when it has to be initialized) is a bit dangerous code.
Also lambdas are pretty well optimized in the modern compilers, it is going to be inlined.
For better readability I can create separate variable to store a lambda result.

Reply:
You can create the variable with initialization from the `default`? Why would it need to be uninitiated at the beginning?

Anyway, this is a small syntax issue; fine with me if you feel this is better.